### ISSUE TITLE
Validation failed in API: kind is not a valid option when editing a secret or configMap

### ISSUE THREAD

**Issue by blackholegalaxy** [rancher/rancher#18110]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
In a Google Cloud Kubernetes cluster, create a secret containing some information. Here is our gitlab config secret:

Key: **GITLAB_OMNIBUS_CONFIG**

Content:
```
external_url 'https://git.mysite.com';
nginx['ssl_certificate'] = '/var/opt/ssl/gitlab/tls.crt';
nginx['ssl_certificate_key'] = '/var/opt/ssl/gitlab/tls.key';
gitlab_rails['registry_enabled'] = true;
registry_external_url 'https://registry.mysite.com';
nginx['redirect_http_to_https'] = true;
registry_nginx['ssl_certificate'] = '/var/opt/ssl/registry/tls.crt';
registry_nginx['ssl_certificate_key'] = '/var/opt/ssl/registry/tls.key';
gitlab_rails['backup_path'] = '/var/opt/backups';
gitlab_rails['backup_upload_remote_directory'] = 'gitlab-backup';
gitlab_rails['backup_upload_connection'] = {'provider' => 'Google', 'google_storage_access_key_id' => 'my-key', 'google_storage_secret_access_key' => 'my-secret'};
prometheus_monitoring['enable'] = false;
gitlab_rails['gitlab_shell_ssh_port'] = 32671;
gitlab_rails['backup_keep_time'] = 604800;
gitlab_rails['monitoring_whitelist'] = ['127.0.0.0/8', '10.0.0.0/8'];
```

Creation works.

Editing this secret afterward then produces the error. The secret is valid and my gitlab instance starts with no problem.

Please note it also produces the error when defining/editing configMap.

**Result:**

`Validation failed in API`

**Other details that may be helpful:**

![image](https://user-images.githubusercontent.com/11064123/52782088-2fc26f80-301c-11e9-8a36-fef7a1f43691.png)

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI):
```
Rancher | v2.1.6
User Interface | v2.1.19
Helm | v2.10.0-rancher5
Machine | v0.15.0-rancher1-1
```

- Installation option (single install/HA): **Single**

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): **GCE**
- Machine type (cloud/VM/metal) and specifications (CPU/memory): **n1-standard-4**
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.5", GitCommit:"32ac1c9073b132b8ba18aa830f46b77dcceb0723", GitTreeState:"clean", BuildDate:"2018-06-21T11:46:00Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"11+", GitVersion:"v1.11.6-gke.6", GitCommit:"a2fae0aa5d813f814d667711324501312b76e64f", GitTreeState:"clean", BuildDate:"2019-01-18T19:40:57Z", GoVersion:"go1.10.3b4", Compiler:"gc", Platform:"linux/amd64"}
```

- Docker version (use `docker version`):

```
Default on v1.11.6-gke.6
```

**Comment by charlesread** [comment]
I'm experiencing this issue as well with a cloned secret.

**Comment by blackholegalaxy** [comment]
Indeed it was a cloned secret in my case too! Good clue.

**Comment by steveabbitt** [comment]
I can verify seeing this when trying to clone a secret.

**Comment by superseb** [comment]
This should be fixed in 2.2: https://github.com/rancher/rancher/issues/16219
